tests/flow/test_list.py

```diff
@@ -143,3 +143,633 @@ def test03_null_list_function_inputs(self):
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        query = """MATCH p=()-[]->() RETURN toBooleanList(p)"""
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        query = """MATCH p=()-[]->() RETURN toBooleanList([p])"""
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        # Test of type mismatch input
+        try:
+            redis_graph.query("RETURN toBooleanList(true)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Boolean", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList(7.05)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Float", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList(7)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Integer", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList('abc')")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was String", str(e))
+
+        try:
+            query = """CREATE (a:X)"""
+            redis_graph.query(query)
+            query = """MATCH (n) RETURN toBooleanList(n)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Node", str(e))
+            query = """MATCH (a:X) DELETE a"""
+            redis_graph.query(query)
+
+        try:
+            query = """CREATE (a:X), (b:Y)"""
+            redis_graph.query(query)
+            query = """MATCH (a:X), (b:Y) create (a)-[r:R]->(b)"""
+            redis_graph.query(query)
+            query = """MATCH ()-[e]->() RETURN toBooleanList(e)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Edge", str(e))
+            query = """MATCH (a:X),(b:Y) DELETE a, b"""
+            redis_graph.query(query)
+
+        try:
+            query = """CREATE (a:X), (b:Y)"""
+            redis_graph.query(query)
+            query = """MATCH (a:X), (b:Y) CREATE (a)-[r:R]->(b)"""
+            redis_graph.query(query)
+            query = """MATCH p=()-[]->() RETURN toBooleanList(p)"""
+            actual_result = redis_graph.query(query)
```

Comment:
consider compacting 3 queries into a single one

Reply:
Hi, compacting this query I'll get the error reported in #2465
Is there any other alternative?
```
127.0.0.1:6379> GRAPH.QUERY g "CREATE p=(a:X)-[r:R]->(b:Y) RETURN toBooleanList(p)"
(error) _AR_EXP_UpdateEntityIdx: Unable to locate a value with alias p within the record
```